core/src/main/java/org/apache/iceberg/hadoop/SerializableConfiguration.java

```diff
@@ -26,13 +26,18 @@
 
 /** Wraps a {@link Configuration} object in a {@link Serializable} layer. */
 public class SerializableConfiguration implements Serializable {
+  private static final long serialVersionUID = -8840976521081151175L;
 
   private transient Configuration hadoopConf;
 
   public SerializableConfiguration(Configuration hadoopConf) {
     this.hadoopConf = hadoopConf;
   }
 
+  public SerializableConfiguration(Object hadoopConf) {
+    this.hadoopConf = (Configuration) hadoopConf;
+  }
```

Comment:
What's the point here? When there's no Hadoop one classpath then it will blow up no matter what, right?
Additionally explicit casts are just brittle. This question applies to all other such places where `Object` is passed.
In Flink this is solved in a way that Hadoop specific class usages are protected with `isHadoopCommonOnClasspath` and that's it, works like charm.